include/hip/devicelib/double_precision/dp_math.hh

```diff
@@ -25,28 +25,86 @@
 
 #include <hip/devicelib/macros.hh>
 
+#if defined __has_builtin && __has_builtin(__builtin_acos)
```

Comment:
When would these be false?

Reply:
The guards are for just in case. It's unlikely that Clang would drop the built-ins, I think.